### PR TITLE
fix(ext/console): fix error when logging a proxied Date

### PR DESCRIPTION
**Pull request by liruifengv**
fix https://github.com/denoland/deno/issues/14164

### CODE REVIEW

**Comment by kt3k** [comment]
Node.js seems swapping logging target from proxy itself to proxy's target here https://github.com/nodejs/node/blob/e58c7ccc97d48aa70bc716f7902caedcff2a429c/lib/internal/util/inspect.js#L789

I think we should follow this pattern instead of branching in each `inspectXYZ` function

**Comment by kt3k** [comment]
We already get `proxyDetails` here 
 https://github.com/denoland/deno/blob/9a9dd12253a5c26df16dd14d61732110e1d1eb72/ext/console/02_console.js#L635-L638 but we don't use it unless `showProxy` option is set. Node.js uses proxy's target object as inspection target by default

File: ext/console/02_console.js [hunk 1]
```diff
@@ -674,7 +674,9 @@
           return handleCircular(value, cyan);
         }
 
-        return inspectObject(value, inspectOptions, proxyDetails);
+        const targetObject = proxyDetails ? proxyDetails[0] : value
```
Comment (liruifengv) [inline ext/console/02_console.js:677]: I used `proxyDetails` if it existed.
But the test `inspectProxy` seems failed.

File: ext/console/02_console.js [hunk 2]
```diff
@@ -674,7 +674,10 @@
           return handleCircular(value, cyan);
         }
 
-        return inspectObject(value, inspectOptions, proxyDetails);
+        return inspectObject(
+          proxyDetails ? proxyDetails[0] : value,
```
Comment (liruifengv) [inline ext/console/02_console.js:678]: I used `proxyDetails` if it existed.
But the test `inspectProxy` seems failed.

File: ext/console/02_console.js [hunk 3]
```diff
@@ -1284,15 +1283,9 @@
     } else if (ObjectPrototypeIsPrototypeOf(DatePrototype, value)) {
       return inspectDate(value, inspectOptions);
     } else if (ObjectPrototypeIsPrototypeOf(SetPrototype, value)) {
-      return inspectSet(
-        proxyDetails ? proxyDetails[0] : value,
```
Comment (liruifengv) [inline ext/console/02_console.js:1288]: Maybe i should write code like this in Date branch?

Comment (kt3k) [reply ext/console/02_console.js:1288]: I'd accept that change too.

Comment (liruifengv) [reply ext/console/02_console.js:1288]: I have modified it according to this.

**Review by kt3k — APPROVED** [review]
LGTM. Thanks for the fix!

**Pull request merged**